### PR TITLE
Add logic to section router

### PR DESCRIPTION
**Pull request by lfryemason**
Adds the router logic for the new section navigation. Including:
* Moving TeacherDashboardPaths into a constant in a shared file
* Removes the `nav_test_v2` and instead shows the navigation bar as part of the new router
* Continues to redirect unknown paths to progress page
* Move empty section logic from `TeacherDashboard` to `EmptySection`

https://github.com/user-attachments/assets/3f9ad1d7-e3a9-4f01-80c1-26a6bc12c479

TODO: 
* Investigate whether we still need other logic in TeacherDashboard

## Links
[TEACH-1233](https://codedotorg.atlassian.net/browse/TEACH-1233)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/teacherDashboard/teacherNavigation/TeacherDashboardPaths.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  progress: '/progress',
+  textResponses: '/text_responses',
+  assessments: '/assessments',
+  projects: '/projects',
+  stats: '/stats',
+  manageStudents: '/manage_students',
+  loginInfo: '/login_info',
+  standardsReport: '/standards_report',
+  aiTutorChatMessages: '/ai_tutor',
+  navTestV2: '/nav_test_v2',
```
Comment (kobryan0619) [inline apps/src/templates/teacherDashboard/teacherNavigation/TeacherDashboardPaths.tsx]: Do you think we need this any more?

File: apps/src/templates/teacherDashboard/TeacherDashboard.jsx [hunk 2]
```diff
@@ -87,56 +81,23 @@ function TeacherDashboard({
     }
   });
 
-  const [teacherNavV2Enabled, setTeacherNavV2Enabled] = useState(false);
-
-  useEffect(() => {
-    setTeacherNavV2Enabled(
-      DCDO.get('teacher-local-nav-v2', false) ||
```
Comment (kobryan0619) [inline apps/src/templates/teacherDashboard/TeacherDashboard.jsx:94]: It looks like you got rid of the DCDO flag altogether here.  Was that your intention?

**Review by kobryan0619 — APPROVED** [review]
Looks good - a few comments/questions.

File: apps/src/templates/teacherDashboard/teacherNavigation/TeacherDashboardPaths.tsx (hunk 1 above)
Comment (lfryemason) [reply apps/src/templates/teacherDashboard/teacherNavigation/TeacherDashboardPaths.tsx]: Good call, we do not.

File: apps/src/templates/teacherDashboard/TeacherDashboard.jsx (hunk 2 above)
Comment (lfryemason) [reply apps/src/templates/teacherDashboard/TeacherDashboard.jsx:94]: Yes, we are instead relying on [this](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/sites/studio/pages/teacher_dashboard/show.js#L101) use of the flag that was added previously

**Pull request merged**